cep-base-url.md

```diff
@@ -0,0 +1,45 @@
+<table>
+<tr><td> Title </td><td> Hosting repodata.json and packages separately by adding a base_url property. </td>
+<tr><td> Status </td><td> Draft </td></tr>
+<tr><td> Author(s) </td><td> Daniel Holth &lt;dholth@anaconda.com&gt;</td></tr>
+<tr><td> Created </td><td> Aug 24, 2023</td></tr>
+<tr><td> Updated </td><td> Aug 24, 2023</td></tr>
+<tr><td> Discussion </td><td>  </td></tr>
+<tr><td> Implementation </td><td>  </td></tr>
+</table>
+
+## Abstract
+
+Conda packages need to be stored in the same directory as repodata.json.
+
+This can be inconvenient especially when generating alternative repodata.json for the same set of packages.
+
+## Specification
+
+A minimal `repodata.json` looks like this.
+
+```
+{"info": {"subdir": "linux-64"},
+ "packages": {},
+ "packages.conda": {"package-name.conda":{...}},
+ "removed": [],
+ "repodata_version": 1}
+```
+
+Add `base_url` to the `info` object.
+
+```
+{"info": {"subdir": "...", "base_url":"https://repo.anaconda.com/repo/main/linux-64/"},
+ "packages": {},
+ "packages.conda": {"package-name.conda":{...}},
+ "removed": [],
+ "repodata_version": 1}
```

Comment:
On the other hand bumping the version would probably not cause problems, since we would continue to serve repodata 1 on the normal channels and serve repodata 2 on any "we need base_url" channels.

Reply:
I wonder if conda would work properly if the package keys were sha hashes so that you could point the whole thing at a content-addressed store instead of a server with normal filenames.

Reply:
> Maybe so. Did we bump it when we added packages.conda?

Nope. See https://github.com/conda/conda/pull/8639.

I don't think it even came up in the conversation. The rationale might have been that it's backwards compatible because it only adds one field, so v1-compatible clients (given that there's no validation of extra keys) would just ignore it. I would have bumped it though so new clients with .conda-compatible features know that they can attempt to find it 🤷

Reply:
> I wonder if conda would work properly if the package keys were sha hashes so that you could point the whole thing at a content-addressed store instead of a server with normal filenames.

This is interesting, and probably worth tackling in this CEP, but under a new name à la "SHA-indexed repodata format", where `base_url` happens to be one of the changes proposed in the CEP (because we need a base URL to fetch those hashes from).

It would certainly help with the OCI mirror case.